src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -378,44 +385,89 @@ public static void main(String[] args) {
-                    allValid = false;
+                    invalidCount++;
                 }
                 date = Calendar.getInstance().getTime();
                 logger.info("---> completed " + format.format(date) + ".");
-            }
+            } else
+                skipCount++;
+
             dataToBeValidated.close();
         }
 
-        if (!allValid) {
-            logger.info("At least one KB was invalid" + (skipCount > 0 ? " (" + skipCount + " skipped)." : "."));
-            // Return a failure code if anything fails to validate.
-            System.exit(ReturnCode.VALIDATION_ERROR.ordinal());
+        final ReturnCode returnCode = displaySummary(fileNum, invalidCount, skipCount);
+        System.exit(returnCode.ordinal());
+    }
+
+    // Check for hypotheses > 5MB
+    private static boolean checkHypothesisSize(File fileToValidate) {
+        try {
+            final Path path = Paths.get(fileToValidate.toURI());
+            final long fileSize = java.nio.file.Files.size(path);
+            if (fileSize > 1024 * 1024 * 5) { // 5MB
+                logger.warn("---> Hypothesis KB " + fileToValidate + " is more than 5MB (" + fileSize + " bytes); skipping.");
+                return true;
+            } else {
+                return false;
+            }
+        } catch (IOException ioe) {
+            logger.warn("---> Could not determine size for hypothesis KB " + fileToValidate + "; skipping.");
+            return true;
+        }
+    }
+
+    // Load the model, or fail trying
+    private static boolean loadFile(OntModel dataToBeValidated, File fileToValidate) {
+        dataToBeValidated.addLoadedImport(INTERCHANGE_URI);
+        dataToBeValidated.addLoadedImport(AIDA_DOMAIN_COMMON_URI);
+        try {
+            loadModel(dataToBeValidated, Files.asCharSource(fileToValidate, Charsets.UTF_8));
+        } catch (RuntimeException rte) {
+            logger.warn("---> Could not read " + fileToValidate + "; skipping.");
+            return true;
+        }
+        return false;
+    }
+
+    // Display a summary to the user
+    private static ReturnCode displaySummary(int fileCount, int invalidCount, short skipCount) {
+        if (fileCount == 1) { // Tune output to single-file validation
+            if (skipCount == 1) {
+                logger.info("No validation was performed.");
+            } else {
+                logger.info("The submitted KB was " + (invalidCount == 1 ? "invalid." : "valid."));
+            }
+        } else if (invalidCount > 0) { // Multiple file validation
+            final int validCount = fileCount - invalidCount - skipCount;
+            logger.info("Summary: " + validCount + " valid KB(s), " + invalidCount + " invalid KB(s)" +
+                    (skipCount > 0 ? ", " + skipCount + " skipped." : "."));
+        } else {
+            logger.info("All tested KBs were valid" + (skipCount > 0 ? " (" + skipCount + " skipped)." : "."));
```

Comment:
This seems overly complex. Why don't we have the single-file and multi-file outputs look the same? Couldn't we use the summary for every condition?

Reply:
The reason it's broken out this way is because it bothered me that you'd validate one KB (in this case a hypothesis) and it would say, "All KBs are valid (1 skipped)" if it was too big.  That's a misleading message.  Same thing if you validate one file and it can't be found or read. 
 Furthermore, if you're validating a directory of 10,000 KBs, just seeing, "at least one KB was invalid" isn't particularly helpful.  So I broke them into two cases (essentially `-f` and `-d`) and tracked the *number* of invalid KBs, not just if something had failed, which dated back to the Kotlin version of the validator.

So overall I think this change improves the user interface, at the cost of complexity-- which is why I broke it out into its own method.

Reply:
As we discussed in Teams, I improved summary output by including a report-style summary that looks the same whether you have single- or multiple-file validation.  I also included skipped non-ttl's in the skipCount to reflect better which files were skipped.